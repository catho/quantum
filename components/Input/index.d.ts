import { InputHTMLAttributes, MouseEventHandler, ChangeEventHandler, FC } from 'react';
import { Validations } from '../Form';

export type MaskFunction = (rawValue: string) => string[];
export type Mask = boolean | RegExp | string | MaskFunction;
export type Validate =
  | typeof Validations.Required
  | typeof Validations.CPF
  | typeof Validations.CEP
  | typeof Validations.Date
  | typeof Validations.MinLength
  | typeof Validations.MaxLength
  | typeof Validations.Email;
export type CustomValidate =
  | ((param?: { value: string }) => string)
  | { validate: Validate; error: string };

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  value?: string;
  label?: string;
  helperText?: string;
  descriptionLabel?: string;
  required?: boolean;
  placeholder?: string;
  type?: 'email' | 'text' | 'tel' | 'number' | 'password' | 'search';
  error?: string | false;
  id?: string;
  name?: string;
  validate?: Validate | CustomValidate | Array<Validate | CustomValidate>;
  mask?: Mask | Mask[];
  onClean?: MouseEventHandler<HTMLInputElement>;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  minLength?: number;
  maxLength?: number;
  skin?: 'default' | 'dark';
  theme?: {
    spacing?: {};
    colors?: {};
    baseFontSize?: number;
  };
}

declare const Input: FC<InputProps> & {
  CEP: FC<InputProps>;
  CNPJ: FC<InputProps>;
  CPF: FC<InputProps>;
  Date: FC<InputProps>;
  Password: FC<InputProps>;
  Phone: FC<InputProps>;
};

export default Input;
