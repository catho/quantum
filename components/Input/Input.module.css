.wrapper {
    position: relative;
}

.description-label {
    cursor: text;
    display: block;
    font-size: calc(var(--qtm-base-font-size) * 0.875);
    padding: 0;
    color: inherit;
}

.input-icon {
    cursor: pointer;
    position: absolute;
    right: var(--qtm-spacing-medium);
    bottom: calc(var(--qtm-spacing-xsmall) * 1.25);
    width: calc(var(--qtm-base-font-size) * 1.5);

    color: inherit;
}

.search-icon {
    left: var(--qtm-spacing-medium);
}

.error-icon-default {
    color: var(--qtm-colors-error-700);
}