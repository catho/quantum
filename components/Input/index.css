@import "./Input.module.css";
@import "./sub-components/HelperText.module.css";
@import "./sub-components/InputErrorMessage.module.css";
@import "./sub-components/InputLabel.module.css";
@import "./sub-components/TextInput.module.css";