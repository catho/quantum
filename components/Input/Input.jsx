import PropTypes from 'prop-types';
import styled from 'styled-components';
import React from 'react';
import MaskedInput from 'react-text-mask';

import { FieldGroup, uniqId } from '../shared';
import Icon from '../Icon';
import InputTypes from './InputTypes';
import {
  TextInput,
  InputLabel,
  HelperText,
  RequiredMark,
  InputErrorMessage,
} from './sub-components';
import { spacing, colors, baseFontSize } from '../shared/theme';

const ID_GENERATOR = uniqId('input-');

const InputIcon = styled(Icon)`
  cursor: pointer;
  position: absolute;
  ${({
    theme: {
      spacing: { xsmall, medium },
    },
  }) => `
    right: ${medium}px;
    bottom: ${xsmall * 1.25}px;
    width: ${baseFontSize * 1.5}px;
  `}
`;

const InputSearchIcon = styled(InputIcon).attrs({ name: 'search' })`
  pointer-events: none;
  ${({
    theme: {
      spacing: { medium },
    },
  }) => `left: ${medium}px;`}
`;

const InputErrorIcon = styled(InputIcon).attrs({ name: 'error' })`
  ${({
    theme: {
      colors: {
        error: { 700: error700 },
      },
    },
  }) => `color: ${error700};`}
`;

const DescriptionLabel = styled.span`
  cursor: text;
  display: block;
  ${({ theme: { baseFontSize: fontSize } }) => `
    font-size: ${fontSize * 0.875}px;
    padding: 0;
  `}
`;

const InputWrapper = styled.div`
  position: relative;
`;

InputIcon.displayName = 'InputIcon';
InputSearchIcon.displayName = 'InputSearchIcon';
InputErrorIcon.displayName = 'InputErrorIcon';
DescriptionLabel.displayName = 'DescriptionLabel';

/** A text field component to get user text data */
class Input extends React.Component {
  static CEP = InputTypes.CEP;

  static CNPJ = InputTypes.CNPJ;

  static CPF = InputTypes.CPF;

  static Date = InputTypes.Date;

  static Phone = InputTypes.Phone;

  static Password = InputTypes.Password;

  constructor(props) {
    super(props);

    const { type, id, value } = props;

    this.state = {
      type,
      currentValue: value,
      hasDefaultValue: value !== null && value[0],
    };

    this._id = id || ID_GENERATOR.next().value;
  }

  componentDidUpdate = () => {
    const { currentValue } = this.state;
    const { value } = this.props;

    if (currentValue !== value) {
      this.setState({ currentValue: value });
    }
  };

  onChangeInput = ev => {
    const { onChange } = this.props;
    const inputValue = ev.currentTarget.value;
    this.setState({
      currentValue: inputValue,
      hasDefaultValue: false,
    });

    onChange(ev);
  };

  _changeType = type => {
    this.setState({ type });
  };

  _toggleInputType = () => {
    const { type } = this.state;
    this._changeType(type === 'text' ? 'password' : 'text');
  };

  render() {
    const {
      label,
      error,
      type: typeProp,
      descriptionLabel,
      helperText,
      required,
      onClean,
      theme,
      disabled,
      ...rest
    } = this.props;
    const { currentValue, hasDefaultValue } = this.state;
    const { type: typeState } = this.state;

    const _isSearchType = typeProp === 'search';
    const _isPassword = typeProp === 'password';
    const hasValue = currentValue && currentValue[0];
    const _hasIcon = error || _isPassword || hasValue;
    const hasLabel = !!label;
    const onCleanClick = e => {
      this.setState({
        currentValue: '',
        hasDefaultValue: false,
      });
      onClean(e);
    };

    return (
      <FieldGroup>
        {label && (
          <InputLabel htmlFor={this._id} error={error}>
            {label}
            {required && <RequiredMark>*</RequiredMark>}
          </InputLabel>
        )}
        {descriptionLabel && (
          <DescriptionLabel theme={theme}>{descriptionLabel}</DescriptionLabel>
        )}
        <InputWrapper>
          {_isSearchType && (
            <InputSearchIcon description={descriptionLabel} theme={theme} />
          )}
          <MaskedInput
            {...rest}
            disabled={disabled}
            id={this._id}
            required={required}
            type={typeState}
            value={currentValue}
            onChange={this.onChangeInput}
            render={(ref, props) => (
              <TextInput
                {...props}
                ref={ref}
                error={error}
                hasLabel={hasLabel}
                hasRightIcon={_hasIcon}
                hasLeftIcon={_isSearchType}
                hasDefaultValue={hasDefaultValue}
              />
            )}
          />
          {error && !_isPassword && (
            <InputErrorIcon description={descriptionLabel} theme={theme} />
          )}
          {_isPassword && (
            <InputIcon
              theme={theme}
              name={typeState === 'password' ? 'visibility' : 'visibility_off'}
              description={descriptionLabel}
              onClick={this._toggleInputType}
            />
          )}
          {hasValue && !_isPassword && !error && !disabled && (
            <InputIcon
              theme={theme}
              name="clear"
              description={descriptionLabel}
              onClick={onCleanClick}
            />
          )}
        </InputWrapper>
        {helperText && <HelperText>{helperText}</HelperText>}
        {error && <InputErrorMessage theme={theme}>{error}</InputErrorMessage>}
      </FieldGroup>
    );
  }
}

Input.propTypes = {
  value: PropTypes.string,
  /** Displays a label text that describes the field */
  label: PropTypes.string,
  /** Displays a helper text below the input */
  helperText: PropTypes.string,
  /** Displays a description text below the label */
  descriptionLabel: PropTypes.string,
  required: PropTypes.bool,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf([
    'email',
    'text',
    'tel',
    'number',
    'password',
    'search',
  ]),
  /** Displays an error message and changes border color to error color */
  error: PropTypes.string,
  disabled: PropTypes.bool,
  id: PropTypes.string,
  /**
   * Mask must follow this [rules](https://github.com/text-mask/text-mask/blob/master/componentDocumentation.md#mask)
   */
  mask: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.bool,
    PropTypes.instanceOf(RegExp),
    PropTypes.func,
    PropTypes.string,
  ]),
  onClean: PropTypes.func,
  onChange: PropTypes.func,
  theme: PropTypes.shape({
    spacing: PropTypes.object,
    colors: PropTypes.object,
    baseFontSize: PropTypes.number,
  }),
  minLength: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  maxLength: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Input.defaultProps = {
  error: '',
  disabled: false,
  id: '',
  label: '',
  mask: val => Array(val.length).fill(/./),
  type: 'text',
  value: '',
  helperText: '',
  descriptionLabel: '',
  required: false,
  placeholder: '',
  onClean: () => {},
  onChange: () => {},
  theme: { spacing, colors, baseFontSize },
  minLength: 0,
  maxLength: 255,
};

Input.displayName = 'Input';

export default Input;
