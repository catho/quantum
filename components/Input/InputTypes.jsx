import React from 'react';
// eslint-disable-next-line import/no-cycle
import Input from './Input';
import masks from '../shared/masks';

const InputTypes = {
  CEP: props => <Input mask={masks.cep} {...props} />,
  CNPJ: props => <Input mask={masks.cnpj} {...props} />,
  CPF: props => <Input mask={masks.cpf} {...props} />,
  Date: props => <Input mask={masks.date} {...props} />,
  Phone: props => <Input mask={masks.phone} {...props} />,
  Password: props => <Input type="password" {...props} />,
};

InputTypes.CEP.displayName = 'Input.CEP';
InputTypes.CNPJ.displayName = 'Input.CNPJ';
InputTypes.CPF.displayName = 'Input.CPF';
InputTypes.Date.displayName = 'Input.Date';
InputTypes.Phone.displayName = 'Input.Phone';
InputTypes.Password.displayName = 'Input.Password';

export default InputTypes;
