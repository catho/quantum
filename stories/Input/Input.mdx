import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import { Input, Alert } from '../../components';

import { Header } from '../shared';

import { 
  Default,
  Label,
  Error,
  Disabled,
  DescriptionLabel,
  HelperText,
  Searchable,
  CEP,
  CNPJ,
  CPF,
  Date,
  Phone,
  Password,
  Skin,
  ControlledExample,
} from './Input.stories.jsx';

import EvaluationForm from '../../stories/shared/EvaluationForm'



<Meta title="Forms/Input" component={Input}  />

<Header title="Input">
  The Input component is used to create interactive controls for web-based forms in order to accept data from the user.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import Input from '@catho/quantum/Input';
```

<Canvas>
  <Story of={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Input} />

## <a id="examples"></a>Examples

### With label
<Canvas>
  <Story of={Label} />
</Canvas>

### With error
<Canvas>
  <Story of={Error} />
</Canvas>

### Disabled
<Canvas>
  <Story of={Disabled} />
</Canvas>

### Description label
<Canvas>
  <Story of={DescriptionLabel} />
</Canvas>

### Helper text
<Canvas>
  <Story of={HelperText} />
</Canvas>

### Searchable
<Canvas>
  <Story of={Searchable} />
</Canvas>

### CEP
<Canvas>
  <Story of={CEP} />
</Canvas>

### CNPJ
<Canvas>
  <Story of={CNPJ} />
</Canvas>

### CPF
<Canvas>
  <Story of={CPF} />
</Canvas>

### Date
<Canvas>
  <Story of={Date} />
</Canvas>

### Phone
<Canvas>
  <Story of={Phone} />
</Canvas>

### Password
<Canvas>
  <Story of={Password} />
</Canvas>

### Skin

<Canvas>
  <Story of={Skin} />
</Canvas>

<Alert icon="info" skin="primary">
  The use of dark skin in input should be applied to backgrounds that
  have a high level of contrast for accessibility and reading
  purposes, so use it with a Primary 700 theme or higher in contrast
  level, avoid light backgrounds.
</Alert>

### Controlled example
<Canvas>
  <Story of={ControlledExample} />
</Canvas>

<EvaluationForm componentName="Input" />
