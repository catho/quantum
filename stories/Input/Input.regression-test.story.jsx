import React from 'react';
import { storiesOf } from '@storybook/react';

import Input from '../../components/Input';

storiesOf('Input', module)
  .add('default', () => <Input />)
  .add('search', () => <Input type="search" />)
  .add('error', () => <Input error="message" />)
  .add('disabled', () => <Input disabled />)
  .add('helper', () => <Input helper="message" />);
