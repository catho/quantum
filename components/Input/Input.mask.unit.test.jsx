/* eslint-disable react/display-name */
import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

jest.mock('react-text-mask', () => props => (
  <input type="text" {...{ ...props }} />
));

describe('Input component', () => {
  it('should match snapshots', () => {
    const INPUTS = [
      <Input />,
      <Input value="foo" />,
      <Input label="Text label" />,
      <Input error="Error message" />,
      <Input required />,
      <Input type="search" />,
      <Input disabled />,
      <Input helperText="this is a helper text" />,
      <Input descriptionLabel="this is a description label" />,
      <Input descriptionLabel="this is a description label" type="search" />,
      <Input placeholder="this input has a placeholder" />,
      <Input.CEP />,
      <Input.CNPJ />,
      <Input.CPF />,
      <Input.Date />,
      <Input.Phone />,
      <Input.Password />,
    ];

    INPUTS.forEach(input =>
      expect(renderer.create(input).toJSON()).toMatchSnapshot(),
    );
  });
});
