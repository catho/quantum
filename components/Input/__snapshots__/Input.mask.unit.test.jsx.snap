// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshots 1`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-0"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 2`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  cursor: pointer;
  position: absolute;
  right: 16px;
  bottom: 10px;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-1"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value="foo"
    />
    
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c2"
      description=""
      onClick={[Function]}
      style={
        Object {
          "color": "",
        }
      }
      theme={
        Object {
          "baseFontSize": 16,
          "colors": Object {
            "error": Object {
              "100": "#ffe5e9",
              "300": "#f88e86",
              "500": "#ec584e",
              "700": "#e91b0c",
              "900": "#ac001a",
            },
            "neutral": Object {
              "0": "#ffffff",
              "100": "#f2f2f2",
              "1000": "#000000",
              "300": "#e0e0e0",
              "500": "#999999",
              "700": "#424242",
              "900": "#191919",
            },
            "primary": Object {
              "100": "#E5EDFC",
              "300": "#89AAE7",
              "500": "#0CC0EA",
              "700": "#1250C4",
              "900": "#002F7B",
            },
            "secondary": Object {
              "100": "#F7B0C8",
              "300": "#F278A1",
              "500": "#E91E63",
              "700": "#DE0059",
              "900": "#9E003F",
            },
            "success": Object {
              "100": "#dcedc8",
              "300": "#c9eda2",
              "500": "#7ed321",
              "700": "#3b610f",
              "900": "#004D40",
            },
            "warning": Object {
              "100": "#ffefd6",
              "300": "#ffda6a",
              "500": "#ffc107",
              "700": "#f09100",
              "900": "#d14900",
            },
          },
          "spacing": Object {
            "large": 24,
            "medium": 16,
            "small": 12,
            "xlarge": 32,
            "xsmall": 8,
            "xxlarge": 40,
            "xxsmall": 4,
            "xxxlarge": 48,
            "xxxsmall": 2,
          },
        }
      }
    >
      clear
    </span>
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 3`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  display: block;
  font-weight: bold;
  margin: 8px 0 0;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="input-2"
  >
    Text label
  </label>
  
  <div
    className="c2"
  >
    <input
      id="input-2"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 4`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c3 {
  color: #ac001a;
  margin-top: 8px;
  cursor: text;
  display: block;
  line-height: normal;
  font-size: 14px;
  font-style: italic;
  font-weight: 600;
}

.c2 {
  cursor: pointer;
  position: absolute;
  right: 16px;
  bottom: 10px;
  color: #e91b0c;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-3"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c2"
      description=""
      style={
        Object {
          "color": "",
        }
      }
      theme={
        Object {
          "baseFontSize": 16,
          "colors": Object {
            "error": Object {
              "100": "#ffe5e9",
              "300": "#f88e86",
              "500": "#ec584e",
              "700": "#e91b0c",
              "900": "#ac001a",
            },
            "neutral": Object {
              "0": "#ffffff",
              "100": "#f2f2f2",
              "1000": "#000000",
              "300": "#e0e0e0",
              "500": "#999999",
              "700": "#424242",
              "900": "#191919",
            },
            "primary": Object {
              "100": "#E5EDFC",
              "300": "#89AAE7",
              "500": "#0CC0EA",
              "700": "#1250C4",
              "900": "#002F7B",
            },
            "secondary": Object {
              "100": "#F7B0C8",
              "300": "#F278A1",
              "500": "#E91E63",
              "700": "#DE0059",
              "900": "#9E003F",
            },
            "success": Object {
              "100": "#dcedc8",
              "300": "#c9eda2",
              "500": "#7ed321",
              "700": "#3b610f",
              "900": "#004D40",
            },
            "warning": Object {
              "100": "#ffefd6",
              "300": "#ffda6a",
              "500": "#ffc107",
              "700": "#f09100",
              "900": "#d14900",
            },
          },
          "spacing": Object {
            "large": 24,
            "medium": 16,
            "small": 12,
            "xlarge": 32,
            "xsmall": 8,
            "xxlarge": 40,
            "xxsmall": 4,
            "xxxlarge": 48,
            "xxxsmall": 2,
          },
        }
      }
    >
      error
    </span>
    
  </div>
  
  <span
    className="c3"
  >
    Error message
  </span>
</div>
`;

exports[`Input component should match snapshots 5`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-4"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={true}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 6`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  cursor: pointer;
  position: absolute;
  right: 16px;
  bottom: 10px;
  pointer-events: none;
  left: 16px;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c2"
      description=""
      style={
        Object {
          "color": "",
        }
      }
      theme={
        Object {
          "baseFontSize": 16,
          "colors": Object {
            "error": Object {
              "100": "#ffe5e9",
              "300": "#f88e86",
              "500": "#ec584e",
              "700": "#e91b0c",
              "900": "#ac001a",
            },
            "neutral": Object {
              "0": "#ffffff",
              "100": "#f2f2f2",
              "1000": "#000000",
              "300": "#e0e0e0",
              "500": "#999999",
              "700": "#424242",
              "900": "#191919",
            },
            "primary": Object {
              "100": "#E5EDFC",
              "300": "#89AAE7",
              "500": "#0CC0EA",
              "700": "#1250C4",
              "900": "#002F7B",
            },
            "secondary": Object {
              "100": "#F7B0C8",
              "300": "#F278A1",
              "500": "#E91E63",
              "700": "#DE0059",
              "900": "#9E003F",
            },
            "success": Object {
              "100": "#dcedc8",
              "300": "#c9eda2",
              "500": "#7ed321",
              "700": "#3b610f",
              "900": "#004D40",
            },
            "warning": Object {
              "100": "#ffefd6",
              "300": "#ffda6a",
              "500": "#ffc107",
              "700": "#f09100",
              "900": "#d14900",
            },
          },
          "spacing": Object {
            "large": 24,
            "medium": 16,
            "small": 12,
            "xlarge": 32,
            "xsmall": 8,
            "xxlarge": 40,
            "xxsmall": 4,
            "xxxlarge": 48,
            "xxxsmall": 2,
          },
        }
      }
    >
      search
    </span>
    <input
      id="input-5"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="search"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 7`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      disabled={true}
      id="input-6"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 8`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  color: #ac001a;
  margin-top: 8px;
  cursor: text;
  display: block;
  line-height: normal;
  font-size: 14px;
  font-style: italic;
  font-weight: 600;
  color: #424242;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-7"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  <span
    className="c2"
  >
    this is a helper text
  </span>
  
</div>
`;

exports[`Input component should match snapshots 9`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  cursor: text;
  display: block;
  font-size: 14px;
  padding: 0;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  
  <span
    className="c1"
  >
    this is a description label
  </span>
  <div
    className="c2"
  >
    <input
      id="input-8"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 10`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  right: 16px;
  bottom: 10px;
  pointer-events: none;
  left: 16px;
}

.c1 {
  cursor: text;
  display: block;
  font-size: 14px;
  padding: 0;
}

.c2 {
  position: relative;
}

<div
  className="c0"
>
  
  <span
    className="c1"
  >
    this is a description label
  </span>
  <div
    className="c2"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c3"
      description="this is a description label"
      style={
        Object {
          "color": "",
        }
      }
      theme={
        Object {
          "baseFontSize": 16,
          "colors": Object {
            "error": Object {
              "100": "#ffe5e9",
              "300": "#f88e86",
              "500": "#ec584e",
              "700": "#e91b0c",
              "900": "#ac001a",
            },
            "neutral": Object {
              "0": "#ffffff",
              "100": "#f2f2f2",
              "1000": "#000000",
              "300": "#e0e0e0",
              "500": "#999999",
              "700": "#424242",
              "900": "#191919",
            },
            "primary": Object {
              "100": "#E5EDFC",
              "300": "#89AAE7",
              "500": "#0CC0EA",
              "700": "#1250C4",
              "900": "#002F7B",
            },
            "secondary": Object {
              "100": "#F7B0C8",
              "300": "#F278A1",
              "500": "#E91E63",
              "700": "#DE0059",
              "900": "#9E003F",
            },
            "success": Object {
              "100": "#dcedc8",
              "300": "#c9eda2",
              "500": "#7ed321",
              "700": "#3b610f",
              "900": "#004D40",
            },
            "warning": Object {
              "100": "#ffefd6",
              "300": "#ffda6a",
              "500": "#ffc107",
              "700": "#f09100",
              "900": "#d14900",
            },
          },
          "spacing": Object {
            "large": 24,
            "medium": 16,
            "small": 12,
            "xlarge": 32,
            "xsmall": 8,
            "xxlarge": 40,
            "xxsmall": 4,
            "xxxlarge": 48,
            "xxxsmall": 2,
          },
        }
      }
    >
      search
    </span>
    <input
      id="input-9"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="search"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 11`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-10"
      mask={[Function]}
      onChange={[Function]}
      placeholder="this input has a placeholder"
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 12`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-11"
      mask={
        Array [
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          "-",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
        ]
      }
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 13`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-12"
      mask={
        Array [
          /\\\\d/,
          /\\\\d/,
          ".",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          ".",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          "/",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          "-",
          /\\\\d/,
          /\\\\d/,
        ]
      }
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 14`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-13"
      mask={
        Array [
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          ".",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          ".",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          "-",
          /\\\\d/,
          /\\\\d/,
        ]
      }
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 15`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-14"
      mask={
        Array [
          /\\\\d/,
          /\\\\d/,
          "/",
          /\\\\d/,
          /\\\\d/,
          "/",
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
          /\\\\d/,
        ]
      }
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 16`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-15"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="text"
      value=""
    />
    
    
  </div>
  
  
</div>
`;

exports[`Input component should match snapshots 17`] = `
.c0 {
  position: relative;
  margin-bottom: 20px;
  min-width: 250px;
  width: 100%;
}

.c2 {
  cursor: pointer;
  position: absolute;
  right: 16px;
  bottom: 10px;
}

.c1 {
  position: relative;
}

<div
  className="c0"
>
  
  
  <div
    className="c1"
  >
    <input
      id="input-16"
      mask={[Function]}
      onChange={[Function]}
      placeholder=""
      render={[Function]}
      required={false}
      type="password"
      value=""
    />
    
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c2"
      description=""
      onClick={[Function]}
      style={
        Object {
          "color": "",
        }
      }
      theme={
        Object {
          "baseFontSize": 16,
          "colors": Object {
            "error": Object {
              "100": "#ffe5e9",
              "300": "#f88e86",
              "500": "#ec584e",
              "700": "#e91b0c",
              "900": "#ac001a",
            },
            "neutral": Object {
              "0": "#ffffff",
              "100": "#f2f2f2",
              "1000": "#000000",
              "300": "#e0e0e0",
              "500": "#999999",
              "700": "#424242",
              "900": "#191919",
            },
            "primary": Object {
              "100": "#E5EDFC",
              "300": "#89AAE7",
              "500": "#0CC0EA",
              "700": "#1250C4",
              "900": "#002F7B",
            },
            "secondary": Object {
              "100": "#F7B0C8",
              "300": "#F278A1",
              "500": "#E91E63",
              "700": "#DE0059",
              "900": "#9E003F",
            },
            "success": Object {
              "100": "#dcedc8",
              "300": "#c9eda2",
              "500": "#7ed321",
              "700": "#3b610f",
              "900": "#004D40",
            },
            "warning": Object {
              "100": "#ffefd6",
              "300": "#ffda6a",
              "500": "#ffc107",
              "700": "#f09100",
              "900": "#d14900",
            },
          },
          "spacing": Object {
            "large": 24,
            "medium": 16,
            "small": 12,
            "xlarge": 32,
            "xsmall": 8,
            "xxlarge": 40,
            "xxsmall": 4,
            "xxxlarge": 48,
            "xxxsmall": 2,
          },
        }
      }
    >
      visibility
    </span>
    
  </div>
  
  
</div>
`;
